tests/api.rs

```diff
@@ -90,45 +83,52 @@ fn root_data() {
 #[test]
 fn root_string() {
     let _ = ::env_logger::init();
+
     let expr = r#"
 test "hello"
 "#;
-    let mut vm = make_vm();
     fn test(s: RootStr) -> String {
         let mut result = String::from(&s[..]);
         result.push_str(" world");
         result
     }
+
+    let vm = make_vm();
     vm.define_global("test", {
           let test: fn(_) -> _ = test;
           test
       })
       .unwrap();
-    let result = run_expr(&mut vm, expr);
-    match result {
-        Value::String(s) => assert_eq!(&s[..], "hello world"),
-        x => panic!("Expected string {:?}", x),
+
+    let (value, typ) = Compiler::new().run_expr::<Generic<A>>(&vm, "<top>", expr).unwrap();
```

Comment:
I thought that par of the test was to match on the value stored in `Generic`? Same goes for the `array` test. If not I'll use the specific type though, seeing as it's cleaner!